tests/performance/lookup/lookup.rb

```diff
@@ -0,0 +1,78 @@
+# Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+require 'performance_test'
+require 'app_generator/search_app'
+require 'performance/fbench'
+require 'environment'
+
+class Lookup < PerformanceTest
+
+  def initialize(*args)
+    super(*args)
+  end
+
+  def setup
+    super
+    set_owner("balder")
+  end
+
+  def get_app()
+    SearchApp.new.sd(selfdir + "test.sd").
+                  qrservers_jvmargs("-Xms4g -Xmx4g")
+  end
+
+  def test_dictionary_lookup
+    set_description("Test lookupspeed with btree vs hash.")
+    @graphs = [
+      {
+        :title => "Qps",
+        :x => 'legend',
+        :y => 'qps',
+        :historic => true
+      },
+      {
+        :title => 'Average latency',
+        :x => 'legend',
+        :y => 'latency',
+        :historic => true
+      },
+      {
+        :x => 'legend',
+        :y => 'cpuutil',
+        :historic => true
+      },
+      {
+        :x => 'legend',
+        :y => 'memory.rss',
+        :historic => true
+      }
+    ]
+    num_docs = 1000000
+    num_queries = num_docs
+    upper_limit = num_docs*10
+    keys_per_query = 100
+    num_clients = 16
+    deploy_app(get_app())
+    container = (vespa.qrserver["0"] or vespa.container.values.first)
+    container.execute("g++ -Wl,-rpath,#{Environment.instance.vespa_home}/lib64/ -g -O3 -o #{dirs.tmpdir}/docs #{selfdir}/docs.cpp")
```

Comment:
As a followup, we should create a helper function to compile C++ programs. This is spread across many performance tests now.

Reply:
Agree, I will keep that in mind.